integrationtest/mapper/orm-jakarta-batch/src/test/java/org/hibernate/search/integrationtest/jakarta/batch/massindexing/MassIndexingJobWithCompositeIdIT.java

```diff
@@ -54,30 +52,25 @@
  *
  * @author Mincong Huang
  */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
 @TestForIssue(jiraKey = "HSEARCH-2615")
-public class MassIndexingJobWithCompositeIdIT {
+class MassIndexingJobWithCompositeIdIT {
 
 	private static final LocalDate START = LocalDate.of( 2017, 6, 1 );
 
 	private static final LocalDate END = LocalDate.of( 2017, 8, 1 );
 
-	@ClassRule
-	public static ReusableOrmSetupHolder setupHolder =
-			ReusableOrmSetupHolder.withSingleBackend( BackendConfigurations.simple() );
-	@Rule
-	public MethodRule setupHolderMethodRule = setupHolder.methodRule();
+	@RegisterExtension
+	public static OrmSetupHelper ormSetupHelper =
+			OrmSetupHelper.withSingleBackend( BackendConfigurations.simple() );
 
 	private EntityManagerFactory emf;
 
-	@ReusableOrmSetupHolder.Setup
-	public void setup(OrmSetupHelper.SetupContext setupContext) {
-		setupContext.withAnnotatedTypes( EntityWithIdClass.class, EntityWithEmbeddedId.class )
-				.withProperty( HibernateOrmMapperSettings.INDEXING_LISTENERS_ENABLED, false );
-	}
-
-	@Before
-	public void initData() throws Exception {
-		emf = setupHolder.entityManagerFactory();
+	@BeforeAll
+	void setup() {
+		emf = ormSetupHelper.start().withAnnotatedTypes( EntityWithIdClass.class, EntityWithEmbeddedId.class )
+				.withProperty( HibernateOrmMapperSettings.INDEXING_LISTENERS_ENABLED, false )
+				.setup();
```

Comment:
This is nice, but this test assumes that indexes are empty when test methods get first executed... You broke that assumptions, and we may now have some tests that would succeed even if the mass indexing job did nothing at all: we check whether entities are indexed, and they are! But only because we didn't clear indexes between tests...

If you're going this route, I think it would be safer for the `OrmSetupHelper` to automatically clear data/indexes between tests when used in a `@BeforeAll`... and provide an option to disable data clearing for those few tests that need persistent data across tests.